ospfd/ospf_interface.c

```diff
@@ -99,8 +99,8 @@ int ospf_if_get_output_cost(struct ospf_interface *oi)
 		cost = (uint32_t)((double)refbw / (double)bw + (double)0.5);
 		if (cost < 1)
 			cost = 1;
-		else if (cost > 65535)
-			cost = 65535;
+		else if (cost > UINT32_MAX)
+			cost = UINT32_MAX;
```

Comment:
Isn't `65535` supposed to mean an infinite cost here?

We have this in _lib/libospf.h_:
```c
#define OSPF_OUTPUT_COST_INFINITE          UINT16_MAX
```

Looking at the OSPF RFC, I see that the metric field of Router LSAs contains only 16 bits, so the `cost > 65535` check is necessary to prevent overflows. Also, `cost` is an `uint32_t` variable, so checking if it's bigger than `UINT32_MAX` doesn't make sense.